python/ray/tune/trainable.py

```diff
@@ -116,34 +112,22 @@ def train(self):
         start = time.time()
         result = self._train()
         self._iteration += 1
-        if result.time_this_iter_s is not None:
-            time_this_iter = result.time_this_iter_s
-        else:
-            time_this_iter = time.time() - start
 
-        if result.timesteps_this_iter is None:
-            raise TuneError("Must specify timesteps_this_iter in result",
-                            result)
-
-        self._time_total += time_this_iter
-        self._timesteps_total += result.timesteps_this_iter
-
-        # Include the negative loss to use as a stopping condition
-        if result.mean_loss is not None:
-            neg_loss = -result.mean_loss
+        if result.time_this_iter_s is None:
+            time_this_iter = time.time() - start
         else:
-            neg_loss = result.neg_mean_loss
+            time_this_iter = result.time_this_iter_s
+        self._time_total += time_this_iter
 
         now = datetime.today()
-        result = result._replace(
+        result = result.copy()
+        result.update(
             experiment_id=self._experiment_id,
             date=now.strftime("%Y-%m-%d_%H-%M-%S"),
             timestamp=int(time.mktime(now.timetuple())),
             training_iteration=self._iteration,
-            timesteps_total=self._timesteps_total,
```

Comment:
Let's make sure timesteps_total is still calculated correctly, but it can be optional.